engine/class_modules/warlock/sc_warlock_demonology.cpp

```diff
@@ -250,7 +244,14 @@ struct hand_of_guldan_t : public demonology_spell_t
   {
     demonology_spell_t::impact( s );
 
-    impact_spell->execute();
+    impact_spell->execute();    
+
+    if ( p()->legendary.forces_of_the_horned_nightmare.ok() &&
+         rng().roll( p()->legendary.forces_of_the_horned_nightmare->effectN( 1 ).percent() ) )
+    {
+      p()->procs.horned_nightmare->occur();
+      make_event( *sim, 400_ms, [ this, t = target ] { impact_spell->execute_on_target( t ); } );
```

Comment:
This should be `[ this, t = s->target ]`.

Reply:
`target` and `s->target` can differ when: a) you're dealing with an AoE spell or b) when the spell has travel time and executes on multiple targets in a quick succession

Reply:
Yes, it's most likely fine here but it's a potential error waiting to happen.

Reply:
Understood. Like I said I certainly agree with the change, was just trying to gain an understanding of the situation because I am a bit new to the engine.